### PR TITLE
Replace callbacks with Promises

### PR DESCRIPTION
**Pull request by belochub**

### CODE REVIEW

File: lib/pg.provider.js [hunk 1]
```diff
@@ -692,141 +474,75 @@ class PostgresProvider extends StorageProvider {
...
+      this.systemLogger(err);
+      return client.query('ROLLBACK').then(
+        () => {
+          throw new GSError(errorCodes.INTERNAL_PROVIDER_ERROR, err);
+        },
+        rollbackError => {
+          this.systemLogger(rollbackError);
+          throw new GSError(errorCodes.INTERNAL_PROVIDER_ERROR, rollbackError);
+        }
+      );
```
Comment (lundibundi) [inline lib/pg.provider.js]: Nit: 
Considering this pattern occurs in a few places I think we can extract it in
```js
const gsRollback = (provider, client, err) => {
  provider.systemLogger(err);
  client.query('ROLLBACK').then(
        () => {
          throw new GSError(errorCodes.INTERNAL_PROVIDER_ERROR, err);
        },
        rollbackError => {
          provider.systemLogger(rollbackError);
          throw new GSError(errorCodes.INTERNAL_PROVIDER_ERROR, rollbackError);
        }
      );
}
```

File: lib/cursor.js [hunk 2]
```diff
@@ -303,33 +306,24 @@ class Cursor {
...
   // to handle remaining operations unsupported by current cursor
   //   data - <Array>, rows to date
-  //   callback - <Function>, to be called upon completion
-  //     err - <Error> | <null>
-  //     dataset - <Object[]>
-  //     cursor - <MemoryCursor>
-  continue(data, callback) {
-    new MemoryCursor(data, {
+  // Returns: <Promise>
+  continue(data) {
```
Comment (nechaido) [inline lib/cursor.js]:
```suggestion
  async continue(data) {
```

File: lib/permission.js [hunk 3]
```diff
@@ -94,16 +92,19 @@ SELECT DISTINCT "Application"."Name" FROM "Application"
...
-  provider.pool.query(categoryQuery(), [userId, category, flag], (err, res) => {
-    if (err) {
-      provider.systemLogger(err);
-      callback(err);
-    } else {
-      callback(null, res.rows[0].exists);
-    }
-  });
+  try {
+    const res = await provider.pool.query(categoryQuery(), [
```
Comment (SemenchenkoVitaliy) [inline lib/permission.js]:
```js
    const res = await provider.pool.query(
      categoryQuery(),
      [userId, category, flag]
    );
```
?

Comment (belochub) [reply lib/permission.js]: I don't see the difference, is it just formatting? I didn't format the code manually, it is done automatically by Prettier.

Comment (SemenchenkoVitaliy) [reply lib/permission.js]: Oh, if that was Prettier then this is fine.

**Pull request merged**